File: pet_hospital_backend/src/main/java/com/pethospital/entity/VaccineRecord.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("vaccine_records")
public class VaccineRecord {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long petId;
    private String vaccineName;
    private String vaccineType;
    private String batchNumber;
    private LocalDate vaccinationDate;
    private LocalDate nextDueDate;
    private Long doctorId;
    private Integer status; // 0-未支付，1-待接种，2-已完成
    private String notes;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic(value = "null", delval = "now()")
    private LocalDateTime deletedAt;
    
    // 关联字段
    @TableField(exist = false)
    private Pet pet;
    
    @TableField(exist = false)
    private Employee doctor;
    
    @TableField(exist = false)
    private java.math.BigDecimal price; // 疫苗价格，从库存表关联获取
}

File: pet_hospital_frontend/src/api/vaccineRecord.ts
import request from './index'

export interface VaccineRecord {
    id?: number
    petId: number
    vaccineName: string
    vaccineType?: string
    batchNumber?: string
    vaccinationDate: string
    nextDueDate?: string
    doctorId?: number
    status?: number
    notes?: string
    createdAt?: string
    price?: number
    pet?: {
        name?: string
        species?: string
    }
    doctor?: {
        name?: string
    }
}

export const vaccineRecordApi = {
    list: (params?: { petId?: number }) =>
        request.get<VaccineRecord[]>('/vaccine-records', { params }),

    getById: (id: number) =>
        request.get<VaccineRecord>(`/vaccine-records/${id}`),

    create: (data: VaccineRecord) =>
        request.post<VaccineRecord>('/vaccine-records', data),

    update: (id: number, data: VaccineRecord) =>
        request.put<boolean>(`/vaccine-records/${id}`, data),

    updateStatus: (id: number, status: number) =>
        request.put<boolean>(`/vaccine-records/${id}/status`, null, { params: { status } }),

    delete: (id: number) =>
        request.delete<boolean>(`/vaccine-records/${id}`)
}
